backend/src/main/java/com/soen341/instagram/service/impl/PictureService.java

```diff
@@ -133,4 +150,36 @@ private BufferedImage compressImage(BufferedImage image) {
         compressedImage.createGraphics().drawImage(image, 0, 0, Color.WHITE, null);
         return compressedImage;
     }
+    
+    // like service
+	public int likePicture(final long pictureId) {
```

Comment:
After seeing this method, I understand why you you created 2 functions that gets the picture. An alternative solution that doesn't require the use of 2 functions of `getPictureFromId` is that in your controller, instead of taking the id as long, you can pass it as a string. This is a better solution because it will avoid any type mismatch exception. E.g a user provide a letter as id. The moment that request is made, there's an exception. While if you pass the id as a string, in Luc original methods, he handles the case where the id is not an integer.